src/mongocxx/collection.cpp

```diff
@@ -266,13 +274,13 @@ cursor collection::aggregate(const pipeline& pipeline, const options::aggregate&
         rp_ptr = read_preference()._impl->read_preference_t;
     }
 
-    return cursor(libmongoc::collection_aggregate(_impl->collection_t,
+    return cursor(libmongoc::collection_aggregate(_get_impl().collection_t,
                                                   static_cast<::mongoc_query_flags_t>(0),
                                                   stages.bson(), options_bson.bson(), rp_ptr));
 }
 
 void* collection::implementation() const {
```

Comment:
It is a backdoor into the C driver in case there is something you can't do. We should probably remove it before GA.